src/lib/runner.rs

```diff
@@ -122,18 +122,37 @@ pub fn generate(missile: &Missile, launch_parameters: &LaunchParameter, timestep
 
 
 		let engine_stage;
-
-		if (f64::from(i) * timestep) < missile.timefire0 {
-			a = ((missile.force0 - drag_force) / missile.mass) - gravity;
-			engine_stage = "0";
-		} else if missile.timefire1 != 0.0 && (f64::from(i) * timestep) < missile.timefire1 {
-			a = ((missile.force1 - drag_force) / missile.mass) - gravity;
-			engine_stage = "1";
-		} else {
-			a = (-drag_force / missile.mass_end) - gravity;
-			engine_stage = "-";
+		let mass;
+		let force;
+
+		let burn_0 = 0.0..missile.timefire0;
+		let burn_1 = burn_0.end..missile.timefire1;
```

Comment:
should be something like
let burn_1 = burn_0.end..**(**missile.timefire1**+burn_0.end)**;

if you had missile with
    "timefire0": 10,
    "timefire1": 5,
then the second phase would be skipped

Reply:
I went and manually verified, the range from burn_0.end returns a range similar to:
time_fire = 5
time_fire = 10
0..5 5..10

Reply:
Yea but 
AIM7F should accelerate for 4,5 + 11 seconds

not 4,5 + 6.5

from the log:
**ts(s): 10.9** D(m): 6850 Dt(m): 0    a(m/s²): -13 v(m/s): 604  d(N): 8545  rho: 0.6464 m(kg): 170.3 **s: 1**
**ts(s): 11.0** D(m): 6910 Dt(m): 0    a(m/s²): -50 v(m/s): 599  d(N): 8509  rho: 0.6464 m(kg): 170.1 **s: -**

Reply:
Yup, the range is incorrect on the latest git version

Reply:
Range computes properly now, interpolation doesnt.